client/scripts/build-tests.js

```diff
@@ -8,36 +13,67 @@ const sveltePlugin = require("esbuild-svelte")
 
 build()
 
-async function getTests() {
-  const testFiles = await globby("modules/*/tests/*.ts", {
-    absolute: true
-  })
-  const tests = {}
-  testFiles.forEach(file => {
-    const moduleName = path.basename(path.resolve(path.dirname(file), "../"))
-    const fileNameNoExt = path.basename(file, ".ts")
-    tests[`${moduleName}_${fileNameNoExt}`] = file
-  })
-  return tests
+async function generateMegaBundleEntrypoint() {
+  const sourceFiles = await globby("modules/*/src/**/*.ts")
+  const testFiles = await globby("modules/*/tests/*.ts")
+
+  // compile the entire monorepo into a dynamic import that won't get executed
+  // we do this so that the _entire monorepo_ gets bundled, but won't break anything
+  // this makes sure that code coverage and the like works
+  let entrypoint = "if (!globalThis && 'foo' === undefined) {\n"
```

Comment:
Ah yeah, I should probably explain that. `globalThis` is simple - it's just the modern standard for the global object for Node, workers, web, etc. It's equivalent to `window`, `self`, and `global`. Yes, JS has this problem.

The reason why the expression here is strange is so that optimizers, minifiers, etc. don't dead-code this section. So I chose two things I knew would always be false but was also something that couldn't be analyzed by an optimizer.

Reply:
Okay I see. Yeah please leave a comment explaining that. I also think a variable like `foo` makes it seem like a temporary / debug measure, maybe if it's `'some-string-value'` or `'this-will-always-be-false'` or something it's more clear that it's intended to be there.